src/simplify_cfg.c

```diff
@@ -272,43 +349,31 @@ static void mark_analyzable_variables(struct CfGraph *cfg, bool *did_something)
             if (ins->kind == CF_ADDRESS_OF_VARIABLE)
                 analyzable[find_var_index(cfg, ins->operands[0])] = false;
 
+    bool did_something = false;
     for (int i = 0; i < cfg->variables.len; i++) {
         // Variables cannot become non-analyzable: if it was analyzable before, it
         // is still analyzable now.
         if (cfg->variables.ptr[i]->analyzable)
             assert(analyzable[i]);
 
         if (analyzable[i] && !cfg->variables.ptr[i]->analyzable) {
-            *did_something = true;
+            did_something = true;
             cfg->variables.ptr[i]->analyzable = true;
         }
     }
 
     free(analyzable);
+    return did_something;
 }
 
 static void simplify_cfg(struct CfGraph *cfg)
 {
-    void (*simplifiers[])(struct CfGraph *, bool *) = {
-        remove_unused_variables,
-        mark_analyzable_variables,
-        clean_branches_where_condition_always_true_or_always_false,
-        remove_unreachable_blocks,
-    };
-    int n = sizeof(simplifiers) / sizeof(simplifiers[0]);
-
-    // Run simplifiers one after another until none of them does anything.
-    int nothing_happened_count=0;
-    for (int i = 0; nothing_happened_count < n; i=(i+1)%n) {
-        bool did_something = false;
-        simplifiers[i](cfg, &did_something);
-        if (did_something)
-            nothing_happened_count = 0;
-        else
-            nothing_happened_count++;
-    }
+    do {
+        clean_jumps_where_condition_always_true_or_always_false(cfg);
+    } while (mark_analyzable_variables(cfg));
 
-    //dump_known_bool_values(cfg, determine_known_bool_values(cfg));
+    remove_unreachable_blocks(cfg);
```

Comment:
This has a potential problem: if there are occurrences of `&foo` but they are all in unreachable blocks, then the variable `foo` will never be marked analyzable. In fact, this is the situation in the second function of issue #15.